### PR TITLE
#176 security-setup: disable auth in command line

### PR DESCRIPTION
**Pull request by tymofii-polekhin**
added --no-ssl option to security-setup

### CODE REVIEW

File: security-setup [hunk 1]
```diff
@@ -95,6 +98,10 @@ class Component(object):
         security = self.read_security()
         yield security
         security['security_enabled'] = True
+        if self.args.no_ssl:
+            security.setdefault('enable_ssl', False)
+        else:
+            security.setdefault('enable_ssl', True)
```
Comment (BrianHicks) [inline security-setup]: `security.setdefault('enable_ssl', not self.args.no_ssl)`.

Aside from that, why is this only a default? If someone's passing in this flag, they're explicitly saying to change this setting. `setdefault` won't do that.

File: security-setup [hunk 2]
```diff
@@ -297,11 +304,14 @@ class Consul(Component):
...
-                print('set gossip key')
+            if not self.args.no_ssl:
+                if 'consul_gossip_key' not in config:
+                    config['consul_gossip_key'] = self.random(16)
+                    print('set gossip key')
+                else:
+                    print('gossip key already set')
             else:
-                print('gossip key already set')
+                print('disable ssl security')
```
Comment (BrianHicks) [inline security-setup]: The Consul gossip key doesn't have anything to do with SSL - why is it being modified as part of this PR? If it really does need to modified, that role should be changed to remove the gossip key if `enable_ssl` is not set as well.

File: security-setup [hunk 3]
```diff
@@ -403,10 +413,13 @@ class Zookeeper(Component):
...
-             config.setdefault('zk_consul_ssl', 'true')
-             config.setdefault('zk_consul_ssl_verify', 'false')
-             print('configuring consul ssl defaults')
+        if not self.args.no_ssl:
+            with self.modify_security() as config:
+                 config.setdefault('zk_consul_ssl', 'true')
+                 config.setdefault('zk_consul_ssl_verify', 'false')
+                 print('configuring consul ssl defaults')
+        else:
+            print('disable consul ssl config')
```
Comment (BrianHicks) [inline security-setup]: this doesn't actually disable Consul SSL. You need to set those flags to the correct values. This script needs to be able to be run multiple times to update security preferences.

File: security-setup [hunk 4]
```diff
@@ -415,27 +428,30 @@ class Mesos(Component):  # Mesos should always come after any frameworks
...
+            with self.modify_security() as config:
                 config.setdefault('mesos_credentials', [])
-                credential = {
-                    'principal': config['marathon_principal'],
-                    'secret': config['marathon_secret'],
-                }
-                if credential not in config['mesos_credentials']:
-                    config['mesos_credentials'].append(credential)
-                    print('set auth for Marathon')
+                if 'marathon_principal' in config and 'marathon_secret' in config:
```
Comment (BrianHicks) [inline security-setup]: this is implicit by the class ordering - see the comment on the class definition line. If we're checking for the credentials to be set, however, it needs to *not* fail silently. Can you add a warning, please?

File: security-setup [hunk 5]
```diff
@@ -449,17 +465,18 @@ class Mesos(Component):  # Mesos should always come after any frameworks
...
-                    'principal': config['mesos_follower_principal'],
-                    'secret': config['mesos_follower_secret']
-                })
-            except ValueError:
-                credential_idx = None
-
-            if credential_idx is not None:
-                del config['mesos_credentials'][credential_idx]
-                print('removed follower credentials from credential list - it is now a separate setting.')
+            if not self.args.no_authenticate_followers and not self.args.no_authenticate_frameworks:
```
Comment (BrianHicks) [inline security-setup]: this setting is actually part of a deprecation and should be removed regardless of those flags.

Comment (tymofii-polekhin) [inline security-setup]: this step fails if can't find "mesos_credentials"

File: security-setup (hunk 1 above)
Comment (tymofii-polekhin) [inline security-setup]: fixed

File: security-setup (hunk 2 above)
Comment (tymofii-polekhin) [inline security-setup]: fixed

File: security-setup (hunk 3 above)
Comment (tymofii-polekhin) [inline security-setup]: what does this stand for config.setdefault('zk_consul_ssl', 'true') ?

File: security-setup (hunk 4 above)
Comment (tymofii-polekhin) [inline security-setup]: added

**Pull request merged**